MatrixSDK/MXSession.m

```diff
@@ -5000,6 +5002,12 @@ - (void)decryptEvents:(NSArray<MXEvent*> *)events
                 }
             }
             
+            if (failedEvents.count > 0) {
+                [NSNotificationCenter.defaultCenter postNotificationName:kMXSessionDidFailToDecryptEventsNotification
```

Comment:
There is already class `DecryptionFailureTracker` that deals with decryption failures (and logs them to Posthog analytics). It has some additional checks (i.e. not report if user has not joined the room) and is triggered when rendering the timeline (whereas `MXSession decrypt` could be called in other circumstances as well. I think it would be better to group this functionality instead of reporting failures from multiple places. Could you have a look whether the notification could be posted from there?

Reply:
Hm I understand. I think the truth is that this has not been really consolidated properly before, so different parts of the SDK / app are doing their own tracking. Ideallly the tracking would be handled inside `MXCrypto`, but since we now have two versions, what you did makes sense, especially since the `MXSession` is already collecting `failedEvents` 👍